keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigController.kt

```diff
@@ -54,11 +58,13 @@ class DeliveryConfigController(
       description = "The delivery config. If its `name` matches an existing delivery config the operation is an update, otherwise a new delivery config is created."
     )
     deliveryConfig: SubmittedDeliveryConfig
-  ): DeliveryConfig {
-    validator.validate(deliveryConfig)
-    log.debug("Upserting delivery config '${deliveryConfig.name}' for app '${deliveryConfig.application}'")
-    return repository.upsertDeliveryConfig(deliveryConfig)
-  }
+  ): DeliveryConfig =
+    deliveryConfigProcessors.applyAll(deliveryConfig)
+      .let { processedDeliveryConfig ->
+        validator.validate(processedDeliveryConfig)
```

Comment:
Any reason not to have `DeliveryConfigValidator` implement `DeliveryConfigProcessor` in this PR, even before you break it up?

Reply:
good call, can `DeliveryConfigValidator` implement `DeliveryConfigProcessor` as a whole for now?

Reply:
Yes. I didn't want to over-complicate this PR though.

Reply:
That's fine. I'd encourage you to add a `TODO` in a comment in that case, though.